File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/GetClassMethods.java
package com.onevgo.functions;

import java.lang.reflect.Method;
import java.util.Arrays;

public class GetClassMethods {
    public static Method[] getClassMethods(Class<?> clazz) {
        return clazz.getMethods();
    }

    public static void main(String[] args) {
        Arrays.stream(getClassMethods(myclass.class)).forEach(m -> System.out.println(m.getName()));
    }

    class myclass {
        // constructor
        public myclass() {

        }

        // method 1
        private boolean myfunc1()
        {
            return true;
        }

        // method 2
        public boolean myfunc2()
        {
            return true;
        }
    }
}

File: 01_Language/01_Functions/go/src/functions/get_class_methods.go
package functions

import "reflect"

func GetClassMethods(cls interface{}) []reflect.Method {
	res := make([]reflect.Method, 0, 16)
	cType := reflect.TypeOf(cls)
	for i := 0; i < cType.NumMethod(); i++ {
		res = append(res, cType.Method(i))
	}
	return res
}
